internal/source/git.go

```diff
@@ -13,12 +13,17 @@ import (
 	"github.com/go-git/go-billy/v5/memfs"
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
 	"github.com/go-git/go-git/v5/storage/memory"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
 )
 
 type Git struct {
+	client.Client
+	PodNamespace string
```

Comment:
This is the namespace we're going to read secrets from right? In that case, I'd suggest:
```suggestion
	SecretNamespace string
```